### ISSUE TITLE
Nearby locations empty most of the times

### ISSUE THREAD

**Issue by danielfdsilva** [openaq/openaq.org#145]
We noticed that the "Nearby Locations" will be empty for most of the world except the United States. Not a lot of places have locations within 5km. This can be somewhat off-putting for the users.

We think would be a good idea to change this section to show the nearest X locations regardless of distance. Currently the api doesn't allow to do this so we'd probably need to tweak it a bit.

@jflasher You think that would be feasible?

cc @RocketD0g @olafveerman

**Comment by jflasher** [comment]
Good idea, but not sure exactly how we'd do it. The current query uses postgis to get things within a distance, trying to get the nearest for every location seems like it might be difficult? Looks like http://postgis.net/docs/manual-2.1/ST_ClosestPoint.html might be able to do this.

Another thing we could do is use it as a chance for a CTA if no sensors are nearby. Though I'm not sure what we'd be calling people to do. Buy sensors? Tell their govt to put up sensors?

**Comment by danielfdsilva** [comment]
We already have a CTA, but it's sad that it's an empty section for most of the world

**Comment by danielfdsilva** [comment]
![nearby](https://cloud.githubusercontent.com/assets/1090606/19981383/9f9ebcea-a1f9-11e6-9b81-b7e7a5f1a752.png)

**Comment by danielfdsilva** [comment]
Isn't there a way to get results ordered by distance from a point?

**Comment by jflasher** [comment]
Haha, well I am glad to see my CTA idea was a good one! :) 

What endpoint do you need this on? Is it the `/locations` endpoint? That uses turf since it's a cached response so we'd do it outside postgis.

**Comment by danielfdsilva** [comment]
@jflasher yeah I guess locations would do.
Basically should be something like "order by proximity to this point". Would be great if each result included the distance as well.

**Comment by jflasher** [comment]
How about a new query parameter like `nearest=X` which would give you the X nearest locations and their distance? This should be quick with turf, similar to https://github.com/openaq/openaq-api/blob/develop/api/controllers/locations.js#L111-L129

**Comment by danielfdsilva** [comment]
Sounds like a good idea. How do you pass the `coordinates`?

**Comment by jflasher** [comment]
Good point! How about `coordinates=40.23,34.17&nearest=X`. `nearest` and `radius` are exclusive I guess.

**Comment by danielfdsilva** [comment]
Perfect!

**Comment by jflasher** [comment]
This is now deployed.
